constraint/pkg/client/rego_helpers.go

```diff
@@ -15,151 +14,35 @@ var (
-// ensureRegoConformance rewrites the package path and ensures there is no access of `data`
-// beyond the whitelisted bits. Note that this rewriting will currently modify the Rego to look
-// potentially very different from the input, but it will still be functionally equivalent.
-func (rc *regoConformer) ensureRegoConformance(kind, path, rego string) (string, error) {
-	if rego == "" {
-		return "", errors.New("Rego source code is empty")
-	}
-	module, err := ast.ParseModule(kind, rego)
-	if err != nil {
-		return "", err
-	}
-	if module == nil {
-		return "", errors.New("Module could not be parsed")
-	}
-	if len(module.Imports) != 0 {
-		return "", errors.New("Use of the `import` keyword is not allowed")
-	}
-	// Temporarily unset Package.Path to avoid triggering a "prohibited data field" error
-	module.Package.Path = nil
-	if err := rc.checkDataAccess(module); err != nil {
-		return "", err
-	}
-	module.Package.Path, err = packageRef(path)
-	if err != nil {
-		return "", err
-	}
-	return module.String(), nil
-}
-
-// rewritePackage rewrites the package in a rego module
-func rewritePackage(path, rego string) (string, error) {
-	if rego == "" {
-		return "", errors.New("Rego source code is empty")
-	}
+// parseModule parses the module and also fails empty modules.
+func parseModule(path, rego string) (*ast.Module, error) {
 	module, err := ast.ParseModule(path, rego)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if module == nil {
-		return "", errors.New("Module could not be parsed")
+		return nil, errors.New("Empty module")
 	}
-	module.Package.Path, err = packageRef(path)
-	if err != nil {
-		return "", err
-	}
-	return module.String(), nil
+	return module, nil
 }
 
-// packageRef constructs a Ref to the provided package path string
-func packageRef(path string) (ast.Ref, error) {
+// rewriteModulePackage rewrites the module's package path to path.
+func rewriteModulePackage(path string, module *ast.Module) error {
 	pathParts, err := ast.ParseRef(path)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	packageRef := ast.Ref([]*ast.Term{ast.VarTerm("data")})
-	return packageRef.Extend(pathParts), nil
-}
-
-func makeInvalidRootFieldErr(val ast.Value, allowed map[string]bool) error {
-	if len(allowed) == 0 {
-		return fmt.Errorf("Template is attempting to access `data.%s`. Access to the data document is disabled", val.String())
-	}
-	var validFields []string
-	for field := range allowed {
-		validFields = append(validFields, field)
-	}
-	return fmt.Errorf("Invalid `data` field: %s. Valid fields are: %s", val.String(), strings.Join(validFields, ", "))
-}
-
-var _ error = Errors{}
-
-type Errors []error
-
-func (errs Errors) Error() string {
-	s := make([]string, len(errs))
-	for _, e := range errs {
-		s = append(s, e.Error())
-	}
-	return strings.Join(s, "\n")
-}
-
-// checkDataAccess makes sure that data is only referenced in terms of valid subfields
-func (rc *regoConformer) checkDataAccess(module *ast.Module) Errors {
```

Comment:
IIRC your Rego rewriter reimplements this. This is correct right?

If so, no issue

Reply:
Yeah, it checks usage of 'data' to ensure that it will only reference libs or allowed externs.